src/runtime/http.ts

```diff
@@ -0,0 +1,83 @@
+import { RealTimeConfigInternal } from ".";
+import Logger from "./logger";
+import * as proto from "../proto/server/v1/realtime";
+import axios, { Axios } from "axios";
+import { rejects } from "assert";
+
+interface Channel {
+  channel: string;
+}
+
+interface Subscriptions {
+  devices: string[];
+}
+
+interface Message {
+  id?: string;
+  name: string;
+  data: any;
+}
+
+export class Http {
+  private _config: RealTimeConfigInternal;
+  private logger: Logger;
+  private axios: Axios;
+  constructor(config: RealTimeConfigInternal, logger: Logger) {
+    this._config = config;
+    this.logger = logger;
+    console.log(
+      "URL",
+      `${this._config.url}/v1/projects/${this._config.project}/realtime`
+    );
+    this.axios = axios.create({
+      baseURL: `${this._config.url}/v1/projects/${this._config.project}/realtime`,
+    });
+  }
+
+  async channels(): Promise<Channel[]> {
+    let resp = await this.axios.get("/channels");
+
+    return resp.data?.channels as Channel[];
+  }
+
+  async channel(name: string): Promise<Channel> {
+    let resp = await this.axios.get(`/channels/${name}`);
+    return resp.data as Channel;
+  }
+
+  async channelMessages(name: string, start: string = "0"): Promise<Message[]> {
```

Comment:
If my understanding of channels and subscriptions is correct,  this should be the flow:

```ts
const pubsub = new Http(config, logger)
const channel = pubsub.channel("notifications")
const subscription: Subscription = channel.createSubscription("mySubscriptionName");
// or channel.getSubscription("mySubscriptionName") to resume an existing one if they are stateful 

// streaming pull to consume messages
subscription.subscribe(messageCallback, errorCallback, onEnd);
// OR
subscription.on("message", messageCallback);
subscription.on("error", errCallback);
subscription.on("end", terminalCallback)

// synchronous pull
subscription.getMessages(limit: 100);
```

I previously worked with Node streams that are built using [EventEmitter](https://nodejs.org/api/events.html#class-eventemitter). We can leverage the same - `class Subscription extends EventEmitter {} `

Reply:
I really like the idea of `getMessages(limit...)`. But I think it should be associated with a channel.
```
  let channel = realtime.getChannel("my-chan");

  let messages = await channel.getMessages(start?: "123", limit: 100);
  
```

I'm not sure about the extra `subscription` layer. Messages are linked to a channel, and when a user subscribes they are subscribing to a channel to only receive messages based on the message name in that channel. 
I've based some of the API design on how pusher and Ably support channels:

Ably:
```
const realtime = new Ably.Realtime('xVLyHw.A-pwh7wwYdsBS2h1rfubSBZ2AYqe_SAEc_i9Z792qzriufLWOA');
const channel = realtime.channels.get('elf-nun-hot');
channel.subscribe((message) => {
  alert('Received: ' + message.data);
});
channel.publish('example', 'message data');
```

And pusher:

```
const channel = pusher.subscribe("my-channel");
channel.bind("my-event", (data) => {
  // Method to be dispatched on trigger.
});

pusher.trigger('my-channel', 'my-event', {:message => 'hello world'})
```

I don't want to deviate to much from that pattern since I'm assuming any user that uses our realtime might be familiar with the above designs/